clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2288,39 +2308,6 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::set_cur_input(
         stream->page_size_ = inputs_[input].reader->get_page_size();
     }
 
-    if (inputs_[input].pid != INVALID_PID) {
```

Comment:
Seeing the same changes as the other PRs here that aren't separated: to stack them, you want to have this PR have the prior PR as its base, instead of master.  Then the diff will only be the new changes here.  Please update so it's easier to review.